utils/sl/fake_scons/SCons/Script/__init__.py

```diff
@@ -192,7 +191,28 @@ def Alias(self, *_args, **_kw):
 
     def __getitem__(self, x):
         """Fake __getitem__"""
-        return []
+
+        class myItem():
+            """Fake class for Env variables"""
+
+            def __index__(self):
+                return 0
+
+            def __getitem__(self, x):
+                """Fake __getitem__"""
+
+            def __setitem__(self, x, value):
+                """Fake __setitem__"""
+                return
```

Comment:
Why are these fake classes needed?

Reply:
In general or these ones specifically?  In general so we can test the scons interface with pylint as scons does some odd things, these ones are so we can remove the pylint disabling suppression from https://github.com/daos-stack/daos/blob/master/SConstruct#L332-L333

It's really just making this object behave like a scons Environment so that the checks pass for code we know works.

Reply:
Yes, in general. Thanks for the explanation